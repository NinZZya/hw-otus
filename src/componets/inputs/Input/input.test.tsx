import React from "react";
import renderer from "react-test-renderer";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Input } from "./input";

configure({ adapter: new Adapter() });

describe(`Input render coorectly`, () => {
  it(`Input without props render coorectly`, () => {
    const tree = renderer.create(<Input />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Input with value render coorectly`, () => {
    const tree = renderer.create(<Input value="value" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Input was changed`, () => {
    const onChange = jest.fn();
    const button = shallow(<Input onChange={onChange} />);

    button.simulate("change");
    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
