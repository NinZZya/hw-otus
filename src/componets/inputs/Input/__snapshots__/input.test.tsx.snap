// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render coorectly Input with value render coorectly 1`] = `
<input
  css={
    Object {
      "map": undefined,
      "name": "10feahb",
      "next": undefined,
      "styles": "
  margin: 1rem 0;
  padding: 1rem;
  width: 100%;
  height: 4em;
  border: 1px solid
    rgba(255, 255, 255, 0.1);
  box-sizing: border-box;
  background-color: rgba(255, 255, 255, 0.1);
  color: #ffffff;

  &:focus,
  &:hover,
  &:active,
  &:selected {
    border: 1px solid
      rgba(255, 255, 255, 0.1);
    box-shadow: 0px 0px 5px 1px
      rgba(255, 255, 255, 0.1);
  }
",
      "toString": [Function],
    }
  }
  value="value"
/>
`;

exports[`Input render coorectly Input without props render coorectly 1`] = `
<input
  css={
    Object {
      "map": undefined,
      "name": "10feahb",
      "next": undefined,
      "styles": "
  margin: 1rem 0;
  padding: 1rem;
  width: 100%;
  height: 4em;
  border: 1px solid
    rgba(255, 255, 255, 0.1);
  box-sizing: border-box;
  background-color: rgba(255, 255, 255, 0.1);
  color: #ffffff;

  &:focus,
  &:hover,
  &:active,
  &:selected {
    border: 1px solid
      rgba(255, 255, 255, 0.1);
    box-shadow: 0px 0px 5px 1px
      rgba(255, 255, 255, 0.1);
  }
",
      "toString": [Function],
    }
  }
/>
`;
